### PR TITLE
Error Policy

### PR DESCRIPTION
**Pull request by jbaxleyiii**
In order to be more flexible, and better support SSR and more use cases, this PR introduces a new policy called `ErrorPolicy`. By default, `ErrorPolicy` is set to `none` which is the current behavior of the 1.0 branch. This is done to help quickly start up an app without worrying about errors, as well as maintain an easy migration path to the 2.0.

### ErrorPolicy
ErrorPolicy has three modes:
- `none` allows no errors to come through as results (i.e. doesn't come to `then` or `next`)
- `ignore` allows joint errors and data to come through to `then` and `next` without the errors
- `all` allows errors to come through `then` and `next` on their own, or with data.

*Note* this only applies to `ExecutionResult` errors (i.e. from the server) and client side errors will still throw / fire `error` on observables.

cc @jnwng 

### Checklist:
- [x] Make sure all of the significant new logic is covered by tests

### CODE REVIEW

**Comment by jbaxleyiii** [comment]
We should probably make this a global option as well (on the constructor)

**Comment by jbaxleyiii** [comment]
We need to clean up the `error` vs `errors` API

File: packages/apollo-client/src/core/QueryManager.ts [hunk 1]
```diff
@@ -1085,27 +1097,30 @@ export class QueryManager {
...
-              optimistic: false,
-            });
+            try {
+              // ensure result is combined with data already in store
+              resultFromStore = this.dataStore.getCache().read({
+                variables,
+                query: document,
+                optimistic: false,
+              });
+            } catch (e) {}
```
Comment (NeoPhi) [inline packages/apollo-client/src/core/QueryManager.ts]: Would be good to have a comment indicating why it is okay to ignore this error.

File: packages/apollo-client/src/core/QueryManager.ts [hunk 2]
```diff
@@ -509,6 +509,7 @@ export class QueryManager {
                 lastResult.data &&
                 queryStoreValue.previousVariables === queryStoreValue.variables
               ) {
+                console.log('here');
```
Comment (NeoPhi) [inline packages/apollo-client/src/core/QueryManager.ts]: Stray `console.log`

**Pull request merged**